### PR TITLE
feat: translate package objects

### PR DESCRIPTION
**Pull request by hamnis**
* add package declaration
* replace `package object` with `private object `
* export all members of `private object`

### CODE REVIEW

File: rules/src/main/scala/fix/PackageObjectExport.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+package fix
+
+import scalafix.v1._
+import scala.meta._
+
+class PackageObjectExport extends SemanticRule("PackageObjectExport") {
+  override def fix(implicit doc: SemanticDocument): Patch = {
+    doc.tree.collect { case pkg @ Pkg.Object(_, name, _) =>
```
Comment (ingarabr) [inline rules/src/main/scala/fix/PackageObjectExport.scala:24]: Will this not match package object using the new scala 3 syntax?

Comment (hamnis) [reply rules/src/main/scala/fix/PackageObjectExport.scala:25]: package objects will be removed in Scala3. 

https://docs.scala-lang.org/scala3/reference/dropped-features/package-objects.html

I believe this should match old package objects using the new syntax.

File: rules/src/main/scala/fix/package.scala [hunk 2]
```diff
@@ -32,13 +32,19 @@ package object fix {
       else None
     }
 
+    def splitOn(p: Token => Boolean): Option[(Tokens, Tokens)] = {
+      val idx = tokens.indexWhere(p)
+      if (idx == -1) None else Some(tokens.splitAt(idx))
+    }
```
Comment (ingarabr) [inline rules/src/main/scala/fix/package.scala:38]: Ihttps://github.com/scalameta/scalameta/blob/main/scalameta/tokens/shared/src/main/scala/scala/meta/tokens/Tokens.scala#L73

Comment (ingarabr) [reply rules/src/main/scala/fix/package.scala:38]: how about span?

**Pull request merged**